test/platforms.jl

```diff
@@ -111,3 +111,11 @@ using BinaryBuilderBase: get_march_flags, get_all_arch_names, get_all_march_name
     @test sort(get_all_march_names("x86_64")) == ["avx", "avx2", "avx512", "x86_64"]
     @test sort(get_all_march_names("armv7l")) == ["armv7l", "neonvfpv4"]
 end
+
+@testset "parse" begin
+    @test parse_platform("host") == HostPlatform()
+    @test parse_platform("any") == AnyPlatform()
+    @test parse_platform("x86_64-linux") == Platform("x86_64", "linux")
+    @test parse_platform("x86_64-linux-musl") == Platform("x86_64", "linux", libc="musl")
+    @test parse_platform("x86_64-linux-musl-gcc7") == Platform("x86_64", "linux", libc="musl", libgfortran_version=v"4.0.0")
```

Comment:
Let's do some more varied and useful tests:
```suggestion
    @test parse_platform("i686-w64-mingw32-cxx03") == Platform("i686", "windows"; cxxstring_abi="cxx03")
    @test parse_platform("aarch64-apple-darwin21-libgfortran4-cxx11") == Platform("aarch64", "macos"; os_version=v"21", cxxstring_abi="cxx11", libgfortran_version=v"4")
```